pkg/build/api/v1beta3/types.go

```diff
@@ -167,6 +172,31 @@ type BuildSource struct {
 	SourceSecret *kapi.LocalObjectReference `json:"sourceSecret,omitempty" description:"supported auth methods are: ssh-privatekey"`
 }
 
+// ImageSource describes an image that is used as source for the build
+type ImageSource struct {
+	// From is a reference to an ImageStream tag or a DockerImage to
+	// copy source from
+	From kapi.ObjectReference `json:"from" description:"image reference"`
+
+	// Paths is a list of source and destination paths to copy from the image
+	Paths []ImageSourcePath `json:"paths" description:"paths to copy from image"`
+
+	// PullSecret is a reference to a secret to be used to pull the image from
+	// a registry
+	PullSecret *kapi.LocalObjectReference `json:"pullSecret,omitempty" description:"pull secret for source image"`
+}
+
+// ImageSourcePath describes a path to be copied from a source image and its destination within the build directory.
+type ImageSourcePath struct {
+	// Source is the path of the file or directory inside the image to
+	// copy to the build directory
+	Source string `json:"source" description:"source path (directory or file) inside image"`
+
+	// DestinationPath is the relative directory path within the build directory
+	// where files copied from the image are placed
```

Comment:
Why can't i have destination file?

On Dec 16, 2015, at 4:53 PM, Cesar Wong <notifications@github.com> wrote:

In pkg/build/api/v1beta3/types.go
<https://github.com/openshift/origin/pull/6324#discussion_r47836610>:

> +	// Paths is a list of source and destination paths to copy from the image
> +	Paths []ImageSourcePath `json:"paths" description:"paths to copy from image"`
> +
> +	// PullSecret is a reference to a secret to be used to pull the image from
> +	// a registry
> +	PullSecret *kapi.LocalObjectReference `json:"pullSecret,omitempty" description:"pull secret for source image"`
> +}
> +
> +// ImageSourcePath describes a path to be copied from a source image and its destination within the build directory.
> +type ImageSourcePath struct {
> +	// Source is the path of the file or directory inside the image to
> +	// copy to the build directory
> +	Source string `json:"source" description:"source path (directory or file) inside image"`
> +
> +	// DestinationPath is the relative directory path within the build directory
> +	// where files copied from the image are placed

@smarterclayton <https://github.com/smarterclayton> would SourcePath and
DestinationDir work?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/6324/files#r47836610>.